### PR TITLE
User

### PR DESCRIPTION
**Pull request by AungA07**
update flow of user class and add functions to SqlAccess class (not finished)

### CODE REVIEW

File: database.py [hunk 1]
```diff
@@ -145,7 +204,35 @@ def read_self_table(self):
...
+            # data from that column
+            data = cursor.fetchone()[0]
+            cursor.close()
+            conn.close()
+        return data
+    
+    
+    def admin_get_all_user_data(self, student_id:str, column_name:str):
+        if self.admin_status == 0:
+            raise "Error: user doesn't have admin status"
```
Comment (martincarapia) [inline database.py:223]: Missing the use of a TypeError here, just to stay consistent.

**Review by martincarapia — CHANGES_REQUESTED** [review]

File: database.py [hunk 2]
```diff
@@ -81,6 +81,64 @@ def add_self(self, username:str):
...
+                ''', (student_id, username, admin_status))
+                conn.commit()
+                cursor.close()
+
+                  
+    def remove_user(self, student_id:str):
+        if self.admin_status == 0:
+            raise "Error: user doesn't have admin status"
+        if self.student_id == student_id:
+            raise "Error: can't delete self"
```
Comment (martincarapia) [inline database.py:126]: Same thing here

File: database.py [hunk 3]
```diff
@@ -81,6 +81,64 @@ def add_self(self, username:str):
...
+                INSERT INTO all_users (student_id, username, admin_status, start_time, working_status, total_minutes)
+                VALUES (?, ?, ?, NULL, 0, 0)
+                ''', (student_id, username, admin_status))
+                conn.commit()
+                cursor.close()
+
+                  
+    def remove_user(self, student_id:str):
+        if self.admin_status == 0:
+            raise "Error: user doesn't have admin status"
```
Comment (martincarapia) [inline database.py:124]: Here

File: database.py [hunk 4]
```diff
@@ -81,6 +81,64 @@ def add_self(self, username:str):
         cursor.close()
         conn.close()
 
+    def add_user(self, student_id:str, username:str, admin_status:int):
+        if self.admin_status == 0:
+            raise "Error: user doesn't have admin status"
```
Comment (martincarapia) [inline database.py:86]: here

**Review by martincarapia — CHANGES_REQUESTED** [review]

**Review by martincarapia — APPROVED** [review]
Looks good, thanks

**Review by LuisTheSup — APPROVED** [review]

**Pull request merged**